packages/mobile/src/components/core/ScreenReady.tsx

```diff
@@ -0,0 +1,24 @@
+import type { ReactNode } from 'react'
+
+import Animated, { FadeIn } from 'react-native-reanimated'
+import { usePrevious } from 'react-use'
+
+import { useIsScreenReady } from 'app/hooks/useIsScreenReady'
+
+export const ScreenReady = ({
+  loadingComponent = null,
+  children
+}: {
+  loadingComponent?: ReactNode
+  children: ReactNode
+}) => {
+  const isReady = useIsScreenReady()
```

Comment:
I am a bit worried about having a bunch of isScreenReady timeouts all running separately, maybe we have one singleton and everyone else listens to the state via context?